src/m365/spo/commands/roledefinition/roledefinition-get.ts

```diff
@@ -0,0 +1,86 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { validation } from '../../../../utils';
+import SpoCommand from '../../../base/SpoCommand';
+import { BasePermissions } from '../../base-permissions';
+import commands from '../../commands';
+import { RoleType } from './roleType';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  id: number;
+}
+
+class SpoRoleDefinitionGetCommand extends SpoCommand {
+  public get name(): string {
+    return commands.ROLEDEFINITION_GET;
+  }
+
+  public get description(): string {
+    return 'Gets specified role definition from web by id';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Getting role definition from ${args.options.webUrl}...`);
+    }
+
+    const requestOptions: any = {
+      url: `${args.options.webUrl}/_api/web/roledefinitions(${args.options.id})`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    request
+      .get<any>(requestOptions)
+      .then((response: any): void => {
+
+        const permissions: BasePermissions = new BasePermissions();
+        permissions.high = response.BasePermissions.High as number;
+        permissions.low = response.BasePermissions.Low as number;
+        response["BasePermissionsValue"] = permissions.parse();
```

Comment:
Let's avoid working with indexers because we can't refactor them. Instead, let's define types with the properties that we need to access programmatically so that we can ensure code's integrity

Reply:
✅